pkg/resource/training_job/custom_set_output.go

```diff
@@ -0,0 +1,113 @@
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//     http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+// Use this file if the Status/Spec of the CR needs to be modified after
+// create/describe/update operation
+
+package training_job
+
+import (
+	"context"
+
+	ackv1alpha1 "github.com/aws-controllers-k8s/runtime/apis/core/v1alpha1"
+	svcapitypes "github.com/aws-controllers-k8s/sagemaker-controller/apis/v1alpha1"
+	"github.com/aws/aws-sdk-go/aws"
+	svcsdk "github.com/aws/aws-sdk-go/service/sagemaker"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// customCreateTrainingJobSetOutput sets the resource in TempOutofSync if TrainingJob is
+// in creating state. At this stage we know call to createTrainingJob was successful.
+func (rm *resourceManager) customCreateTrainingJobSetOutput(
+	ctx context.Context,
+	r *resource,
+	resp *svcsdk.CreateTrainingJobOutput,
+	ko *svcapitypes.TrainingJob,
+) (*svcapitypes.TrainingJob, error) {
+	rm.customSetOutput(r, aws.String(svcsdk.TrainingJobStatusInProgress), ko)
+	return ko, nil
+}
+
+// customDescribeTrainingJobSetOutput sets the resource in TempOutofSync if
+// TrainingJob is being modified by AWS. It has an additional check on the debugger status.
+func (rm *resourceManager) customDescribeTrainingJobSetOutput(
+	ctx context.Context,
+	r *resource,
+	resp *svcsdk.DescribeTrainingJobOutput,
+	ko *svcapitypes.TrainingJob,
+) (*svcapitypes.TrainingJob, error) {
+	trainingJobStatus := resp.TrainingJobStatus
+	if resp.DebugRuleEvaluationStatuses != nil && resp.DebugRuleEvaluationStatuses[0].RuleEvaluationStatus != nil {
+		debuggerStatus := resp.DebugRuleEvaluationStatuses[0].RuleEvaluationStatus
```

Comment:
shouldn't this be a loop? if any of the jobs are in InProgress then pass InProgress rather than debuggerStatus because debugger has other statuses like NoIssuesFound where resourceSync should be true

` 'InProgress'|'NoIssuesFound'|'IssuesFound'|'Error'|'Stopping'|'Stopped' `

Reply:
I am not sure I followed. 
1. `any of the jobs are in InProgress` - did you mean training or debugger job ?
2. I have added this status check here - is that not enough ? I just added more conditions to this - 'Stopped' 'Error' 'IssuesFound' based on another comment above

Reply:
wow, yes that makes sense. this would be the case when there are multiple 'debugRuleConfigurations' ?

Reply:
fixing.

Reply:
Pushed the fix, testing in progress. Need to run a full manual check to see debugger re-queuing works as expected.